pkg/scheduler/core/generic_scheduler.go

```diff
@@ -762,6 +763,12 @@ func PrioritizeNodes(
 		return schedulerapi.HostPriorityList{}, errors.NewAggregate(errs)
 	}
 
+	// Run the Score plugins.
+	scoresMap, scoreStatus := framework.RunScorePlugins(pluginContext, pod, nodes)
```

Comment:
Similar to the filter plugins, can we keep the parallelize code here and pass a single node to `RunScorePlugins`?

Reply:
We can, but it will not be as easy to follow as this implementation.  Note that we can't really escape having parallelism inside the framework: the implementation of NormalizeScore will have to employ parallelism across plugins. One argument could be that we should maintain consistency in the place where parallelism is implemented for Score and NormalizeScore, but I don't have strong opinion on this. What do you think?

Reply:
For NormalizeScore, we need to pass all node scores to the plugins that do normalization. Depending on the kind of normalization algorithm we use, we may have parallelism or not. For this particular plugins, I feel if we have the parallelism in one place, it will be easier to follow the code. Of course, this is something we can change in the future. These are implementation details and are not part of our public API.

Reply:
NormalizeScore could parallelize across the plugins similar to what Reduce currently do, but l guess we can leave this discussion to the NormalizeScore PR.

For this one, I prefer to keep it as is for now, I tried to workout a way to move parallelism out, but it is not going to be pretty.

Reply:
That's fine. As I said, we can change this in the future if needed.